lib/cli/src/commands/create_exe.rs

```diff
@@ -419,32 +418,102 @@ impl CreateExe {
+                                        return None;
+                                    }
+                                }
+
+                                if let OperatingSystem::Darwin = target.operating_system {
+                                    if !(p.file_name()?.to_str()?.contains("apple")
+                                        || p.file_name()?.to_str()?.contains("darwin"))
+                                    {
+                                        return None;
+                                    }
+                                }
+
+                                if let OperatingSystem::Linux = target.operating_system {
+                                    if !p.file_name()?.to_str()?.contains("linux") {
+                                        return None;
+                                    }
+                                }
+
+                                Some(p)
+                            })
+                            .next();
+
+                        if let Some(local_tarball) = local_tarball.as_ref() {
+                            let target_file_path = local_tarball
                                 .parent()
-                                .and_then(|parent| Some(parent.join(tarball.file_stem()?)))
-                                .unwrap_or_else(|| tarball.clone());
+                                .and_then(|parent| Some(parent.join(local_tarball.file_stem()?)))
+                                .unwrap_or_else(|| local_tarball.clone());
 
                             let target_file_path = target_file_path
                                 .parent()
                                 .and_then(|parent| Some(parent.join(target_file_path.file_stem()?)))
                                 .unwrap_or_else(|| target_file_path.clone());
 
-                            tarball_dir = target_file_path
-                                .canonicalize()
-                                .unwrap_or_else(|_| target_file_path.clone());
-                            let files = untar(tarball, target_file_path)?;
-                            files.into_iter().find(|f| f.contains(libwasmer_path)).ok_or_else(|| {
-                                anyhow!("Could not find libwasmer for {} target in the fetched release from Github: you can download it manually and specify its path with the --cross-compilation-library-path LIBRARY_PATH flag.", target)})?
+                            let _ = std::fs::create_dir_all(&target_file_path);
+                            let files = untar(local_tarball.clone(), target_file_path.clone())?;
+                            tarball_dir =
+                                target_file_path.canonicalize().unwrap_or(target_file_path);
+                            files.iter().find(|f| f.ends_with("libwasmer.a")).cloned().ok_or_else(|| {
+                                anyhow!("Could not find libwasmer.a for {} target in the provided tarball path (files = {files:#?})", target)})?
```

Comment:
Hang on, didn't we already do this bit?

Reply:
yeah, it's duplicated, I factored it into a function